app/screens/map/mapSelector.js

```diff
@@ -1,11 +1,23 @@
+import pickBy from 'lodash/pickBy';
+
 import { createSelector, createStructuredSelector } from 'reselect';
 
 function unitsSelector(state) {
   return state.data.units;
 }
 
-const markersSelector = createSelector(
+function cityFilterSelector(state) {
+  return state.filters.city;
+}
+
+const filteredUnitsSelector = createSelector(
   unitsSelector,
+  cityFilterSelector,
+  (units, cityFilter) => pickBy(units, unit => cityFilter === '' || unit.city === cityFilter)
```

Comment:
Thought about doing that, but I'm not sure how that'd work once we add more filters. So I figured we should only think about optimizations once we have multiple filters.